frontend/src/components/ChangePasswordModal/index.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import { Form, Input, message, Spin } from "antd";
+import { StyledModal, ModalButton, ModalTitle, ModalLabel } from "./style";
+import api from "../../axios";
+
+const validatePassword = (rule, value, callback) => {
+  var includesSymbol = false;
+  var includesAlpha = false;
+  var includesNum = false;
+
+  var alphas = "qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM";
+  var nums = "1234567890";
+
+  var i;
+  for (i = 0; i < value.length; i++) {
+    if (alphas.includes(value[i])) {
+      includesAlpha = true;
+    } else if (nums.includes(value[i])) {
+      includesNum = true;
+    } else {
+      includesSymbol = true;
+    }
+  }
+
+  if (value && !(value.length >= 8 && includesAlpha && includesNum && includesSymbol)) {
+    callback(
+      "Your password should be at least 8 characters long with at least one letter, one number and one symbol."
+    );
+  } else {
+    callback();
+  }
+};
+
+const ChangePasswordModal = ({ visible, toggleChangePasswordModal }) => {
+  const [loading, setLoading] = useState(false);
+
+  const handleSubmit = (values) => {
+    setLoading(true);
+    api({ sendToken: true })
+      .post("/validate/npassword", values)
+      .then(() => {
+        setLoading(false);
+        message.success("Password changed successfully");
+        toggleChangePasswordModal();
+      })
+      .catch((e) => {
+        setLoading(false);
+        message.error("Error: " + e.response.data);
```

Comment:
Do we want to display the backend's response data to the user?

Reply:
Backend sends explanatory error messages, error field can be arranged but we need to show some message if the request was not successfull